File: Day7/day7.go
/*
Determine total size of each directory
find directories with size of at most 100,000

cd
ls - gives us files and sizes
dir

split each line into parts

if first element is cd, and second is not / we changed 
directory and second element is not .. we need a separate sum

if first element is cd and second is .., then we need to change to 
outer directory

if first element is ls, then the values that 

*/

package main

import (
	"bufio"
	"log"
	"os"
	"strings"
	"fmt"
)

func main() {

	file, err := os.Open("/Users/johnwroge/advent_of_code/Day7/data.txt")
	if err != nil {
		log.Fatal(err)
	}
	defer file.Close()
	lines := [][]string{}
	scanner := bufio.NewScanner(file)

	//iterate while there is text
	for scanner.Scan(){
		//retrieve the current line of text
		line := scanner.Text()
		//split the line of text by spaces into a slice 
		item := strings.Split(line, " ")
		//append slice to 2d slice 
		lines = append(lines, item)
	}

 

}

// func Add(directories []string){
// 	vals := []
// }

func Sum(){
}

File: Day7/day7.ts
import fs from 'fs';

const path = require('path');

const filePath = path.join(__dirname, 'data.txt');
const file = fs.readFileSync(filePath, 'utf-8');

const lines = file.split("\n")

let index: number = 0; 
let files: any = [];
let directory: any = [];

while (index < lines.length){
    //case 1, changing directories
    if (lines[index][1] === "cd"){
        //if changing to home directory, create an empty file
        if (lines[index][2] === "/"){
            var direct:any = [];
            //otherwise if changing into outer directory, remove the last file from the file system
        } else if (lines[index][2] === ".."){
            direct.pop()
            //otherwise, push the current file into the created file
        } else {
            direct.push(lines[index][2])
        }
        //at the end of the loop increment the index
        index += 1
        //check if the command is to list
    } else if (lines[index][1] === "ls"){
        //iterate index(go to next element)
        index += 1
        //loop while the current line is not a command
        while (lines[index][0] != "$"){
            //check if the current element is not 'dir'
            if (lines[index][0] != "dir"){
                //invoke helper function
                Add([lines[index][0], lines[index][1]], direct, lines)
            }
            index += 1; 
        }
    }
}



function Add (file: any[], direct: string[], filesystem: string[]){
    if (direct === []){
        if (filesystem)
    }
}   

function Sum (){

}